### PR TITLE
Wagtail 1.1 compatibility

### PR DESCRIPTION
**Pull request by gasman**

### CODE REVIEW

File: demo/views.py [hunk 1]
```diff
@@ -19,7 +26,7 @@ def search(request):
         search_picks = query.editors_picks.all()
     else:
         search_results = Page.objects.none()
-        search_picks = EditorsPick.objects.none()
+        search_picks = SearchPromotion.objects.none()
```
Comment (kaedroho) [inline demo/views.py]: Shall we chang the variable here to ``search_promotions`` as well?

File: demo/views.py [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
 from wagtail.wagtailcore.models import Page
-from wagtail.wagtailsearch.models import Query, EditorsPick
+from wagtail.wagtailsearch.models import Query
+
+try:
+    # Wagtail >= 1.1
+    from wagtail.contrib.wagtailsearchpromotions.models import SearchPromotion
+except ImportError:
+    # Wagtail < 1.1
+    from wagtail.wagtailsearch.models import EditorsPick as SearchPromotion
```
Comment (kaedroho) [inline demo/views.py]: What's the reason for keeping 1.0 support?

Comment (gasman) [inline demo/views.py]: It means that we can merge this change now, before any 1.1 releases exist (and thus make wagtaildemo usable for development of 1.1 again - currently it isn't).

File: demo/views.py (hunk 1 above)
Comment (gasman) [inline demo/views.py]: I don't think so - the wagtaildemo site front-end still refers to them as Editor's Picks (which is a sensible enough name when used within the context of a search interface - it's only confusing within wagtail admin, where 'editor' has a different meaning), so it makes sense for the template variable to match that.

**Comment by kaedroho** [comment]
Agree with your above comments. Merging now, thanks!

**Pull request merged**